File: C++/Problems/Problems/problem_9.cpp
#include "problem_9.h"

#include <algorithm>
#include <iostream>
#include <ostream>
#include <ranges>


bool problem_9::test() {
    auto n = 121;

    auto expected = true;

    auto result = isPalindrome(n);

    std::cout << result << std::endl;

    return expected == result;
}

bool problem_9::isPalindrome(int x) {
    auto num = std::to_string(x);
    auto len = num.size();
    auto first = num.substr(0, len / 2);
    auto last = num.substr(len - len / 2, len / 2);
    std::ranges::reverse(last);

    return first == last;
}

File: Python/Problems/Problems/problem9.py
from .problem import Problem

class Problem9(Problem):
    def test(self):
        n = 121

        expected = True

        result = self.isPalindrome(n)

        print(result)

        return expected == result

    def isPalindrome(self, x):
        num = str(x)
        return num[::-1] == num